web/src/message_events.js

```diff
@@ -47,6 +47,33 @@ function filter_has_term_type(filter, term_type) {
     );
 }
 
+export function discard_cached_lists_with_term_type(term_type, exclude_current_msg_list) {
+    // Discards cached MessageList and MessageListData which have
+    // `term_type` and `not-term_type`.
+    assert(!term_type.includes("not-"));
+
+    // We loop over rendered message lists and cached message data separately since
+    // they are separately maintained and can have different items.
+    for (const msg_list of message_lists.all_rendered_message_lists()) {
+        if (exclude_current_msg_list && msg_list === message_lists.current) {
```

Comment:
Do we have a use case where we wouldn't want to exclude the current list? It seems like it should basically always be unsafe to discard that one.

Reply:
No we don't have a use case for that, Thanks!